tests/test_utils.py

```diff
@@ -431,3 +432,64 @@ def create_profile_in_trestle_dir(trestle_root: pathlib.Path, catalog_name: str,
     import_ = prof.Import(href=f'{const.TRESTLE_HREF_HEADING}catalogs/{catalog_name}/catalog.json', include_all={})
     profile.imports = [import_]
     fs.save_top_level_model(profile, trestle_root, profile_name, fs.FileContentType.JSON)
+
+
+def get_valid_parts(parts: List[common.Part]) -> List[common.Part]:
+    """Get list of valid parts in list without recursion."""
+    return [part for part in parts if part.id and part.prose] if parts else []
+
+
+def get_total_valid_parts_count(parts: List[common.Part]) -> int:
+    """Get total count of valid parts in parts list."""
+    parts = get_valid_parts(parts)
+    count = len(parts)
+    for part in parts:
+        count += get_total_valid_parts_count(part.parts)
+    return count
+
+
+def parts_equivalent(a: List[common.Part], b: List[common.Part]) -> bool:
+    """Check the total count of valid parts is the same, with recursion."""
+    n_a = get_total_valid_parts_count(a)
+    n_b = get_total_valid_parts_count(b)
+    if n_a != n_b:
+        logging.error(f'count of parts is different: {n_a} vs. {n_b}')
+        return False
+    return True
+
+
+def controls_equivalent(a: cat.Control, b: cat.Control) -> bool:
```

Comment:
I have a strong feeling that comparison of the one instance to another should be placed within the instance's class. There is a `__eq__` method inside of the `BaseModel` that we can override for this purpose and place this code there. 
Of course, given that those files are autogenerated it might be tricky... I would like to know what do you think about it ?

Reply:
This is a relatively crude comparison and I wouldn't want people to think the controls are actually equal - they are just equivalent at a level needed for testing.  So I really think it should be confined to test_utils and only used for testing.